cli/cmd/install.go

```diff
@@ -622,42 +624,48 @@ func (options *installOptions) proxyConfig() *pb.Proxy {
 	}
 }
 
-// linkerdConfigAlreadyExistsInCluster checks the kubernetes API to determine
-// whether a config exists.
+// exitIfClusterExists checks the kubernetes API to determine
+// whether a config exists and exits if it does exist or if an error is
+// encountered.
 //
 // This bypasses the public API so that public API errors cannot cause us to
 // misdiagnose a controller error to indicate that no control plane exists.
-//
-// If we cannot determine whether the configuration exists, an error is returned.
-func linkerdConfigAlreadyExistsInCluster() (bool, error) {
+func exitIfClusterExists() {
 	api, err := k8s.NewAPI(kubeconfigPath, kubeContext)
 	if err != nil {
-		return false, err
+		fmt.Fprintln(os.Stderr, "Unable to build a Kubernetes client to check for configuration. If this expected, use the --ignore-cluster flag.")
+		fmt.Fprintf(os.Stderr, "Error: %s\n", err)
+		os.Exit(1)
 	}
 
 	k, err := kubernetes.NewForConfig(api.Config)
 	if err != nil {
-		return false, err
+		fmt.Fprintln(os.Stderr, "Unable to build a Kubernetes client to check for configuration. If this expected, use the --ignore-cluster flag.")
+		fmt.Fprintf(os.Stderr, "Error: %s\n", err)
+		os.Exit(1)
 	}
 
 	c := k.CoreV1().ConfigMaps(controlPlaneNamespace)
 	if _, err = c.Get(k8s.ConfigConfigMapName, metav1.GetOptions{}); err != nil {
 		if kerrors.IsNotFound(err) {
-			return false, nil
+			return
 		}
 
-		return false, err
+		fmt.Fprintln(os.Stderr, "Unable to build a Kubernetes client to check for configuration. If this expected, use the --ignore-cluster flag.")
+		fmt.Fprintf(os.Stderr, "Error: %s\n", err)
+		os.Exit(1)
 	}
 
-	return true, nil
+	fmt.Fprintln(os.Stderr, "You are already running a control plane. If you would like to ignore its configuration, use the --ignore-cluster flag.")
```

Comment:
Could this message be updated to something more like:

```
Linkerd has already been installed on your cluster in the linkerd namespace. Please run upgrade if you'd like to update this installation. Otherwise, use the --ignore-cluster flag.
```